### PR TITLE
Mdkessler patch 10

### PR DESCRIPTION
**Pull request by mdkessler**
**Did you add yourself as a [contributor](https://github.com/Benjamin-Lee/deep-rules/blob/master/contributors.md) if this is your first contribution?**

- [x] Yes, I added myself or am already a contributor

**Any more details?**

### CODE REVIEW

File: content/09.overfitting.md [hunk 1]
```diff
@@ -1,33 +1,36 @@
...
+Therefore, model performance should be evaluated with a carefully picked panel of relevant metrics that make minimal assumptions about the composition of the testing data [@doi:10.1021/acs.molpharmaceut.7b00578].
+
+When working with biological and medical data, one must also carefully consider potential sources of bias and/or non-independence when defining training and test sets.
+For example, a deep learning model for pneumonia detection in chest X-rays appeared to performed well, but then failed to generalize to outside hospitals [@doi:10.1371/journal.pmed.1002683].
+This resulted from the deep learning model picking up on signal related to which hospital the images were from, and represents a type of artifact or "batch effect" that practitioners must be vigilant towards.
+When dealing with sequence data, holding out test data that are evolutionarily related or that share structural homology to the training data can result in overfitting that is hard to detect due to the inherent relatedness of the partitioned data (cite?).
```
Comment (Benjamin-Lee) [inline content/09.overfitting.md:31]: I could have sworn someone posted a link about this.

File: content/09.overfitting.md [hunk 2]
```diff
@@ -1,33 +1,36 @@
...
-
-The simplest way to combat overfitting is to detect it.
-This can be done by splitting the dataset into three parts: a training set, a tuning set (also commonly called a validation set in the machine learning literature), and a test set.
-By exposing the model solely to the training data during fitting, a researcher can use the model's performance on the unseen test data to measure the amount of overfitting.
+In general, one of the most effective ways to combat overfitting is to detect it in the first place.
+One way to do this is to split the main dataset being worked on into three parts: a training set, a tuning set (also commonly called a validation set in the machine learning literature), and a test set.
+These three partitions allow for the potentially iterative give and take between model learning (on the training set) and updating (on the tuning set) to happen independently of the final model evaluation (on the test set).
+That is, the data used for testing should be "locked away", and should only be used for evaluating the final model after all training and tuning steps are completed (_i.e._ used only once as a final evaluation).
```
Comment (Benjamin-Lee) [inline content/09.overfitting.md:15]:
```suggestion
That is, the data used for testing should be "locked away" and used only once to evaluate the final model after all training and tuning steps are completed.
```

File: content/09.overfitting.md [hunk 3]
```diff
@@ -1,33 +1,36 @@
...
+In other words, it is as if the model is memorizing patterns in the specific data set it is training on rather than learning general relationships that hold across similar datasets.
 This subtle distinction is made clearer by seeing what happens when a model is tested on data to which it was not exposed during training: just as a student who memorizes exam materials struggles to correctly answer questions for which they have not studied, a machine learning model that has overfit to its training data will perform poorly on unseen test data.
 Deep learning models are particularly susceptible to overfitting due to their relatively large number of parameters and associated representational capacity.
-To continue the student analogy, a smarter student has greater potential for memorization than average one and thus may be more inclined to memorize.
+Just as a smarter student may have greater potential for memorization, deep learning models seem more prone to over fitting than traditional machine learning models.
```
Comment (Benjamin-Lee) [inline content/09.overfitting.md:8]:
```suggestion
Just as a smarter student may have greater potential for memorization, deep learning models seem more prone to overfitting than traditional machine learning models.
```

File: content/09.overfitting.md (hunk 1 above)
Comment (Benjamin-Lee) [reply content/09.overfitting.md:31]: #202 is the discussion that made us include that line

File: content/09.overfitting.md [hunk 4]
```diff
@@ -1,33 +1,36 @@
...
-
-The simplest way to combat overfitting is to detect it.
-This can be done by splitting the dataset into three parts: a training set, a tuning set (also commonly called a validation set in the machine learning literature), and a test set.
-By exposing the model solely to the training data during fitting, a researcher can use the model's performance on the unseen test data to measure the amount of overfitting.
+In general, one of the most effective ways to combat overfitting is to detect it in the first place.
+One way to do this is to split the main dataset being worked on into three parts: a training set, a tuning set (also commonly called a validation set in the machine learning literature), and a test set.
+These three partitions allow for the potentially iterative give and take between model learning (on the training set) and updating (on the tuning set) to happen independently of the final model evaluation (on the test set).
```
Comment (SiminaB) [inline content/09.overfitting.md:14]: Not sure about the term "give and take" here. Can we just say "These three partitions allow us to potentially iterate between model learning (on the training set) and updating (on the tuning set) to happen independently of the final model evaluation (on the test set)."

File: content/09.overfitting.md [hunk 5]
```diff
@@ -1,33 +1,36 @@
...
-For example, a deep learning model for pneumonia detection in chest X-rays performed well but failed to generalize to outside hospitals because they were able to detect which hospital the image was from and exploited this information when making predictions [@doi:10.1371/journal.pmed.1002683].
-Similarly, when dealing with sequence data, holding out data that are evolutionarily related or share structural homology to the training data can result in overfitting.
-In these cases, simply holding out test data selected from a random partition of the training data is insufficient.
-The best remedy for confounding variables is to [know your data](#know-your-problem) and to test your model on truly independent data.
+If overfitting is an issue, there are a variety of techniques to reduce overfitting, including data augmentation and regularization techniques such as dropout [@url:http://jmlr.csail.mit.edu/papers/v15/srivastava14a.html] and weight decay [@tag:krogh-weight-decay].
+Another way to reduce overfitting, as described by Chuang and Keiser, is to identify the baseline level of memorization that is occuring by training on data that has it labels randomly shuffled.
```
Comment (SiminaB) [inline content/09.overfitting.md:21]: Is this a kind of permutation test?

File: content/09.overfitting.md [hunk 6]
```diff
@@ -1,33 +1,36 @@
...
+By comparing the model performance with the shuffled data to that achieved with the actual data [@doi:10.1021/acschembio.8b00881], a practitioner can identify overfitting as a model that performs no better on real data, as this suggest that any predictive capacity is not due to data driven signal.
+One important caveat when working with paritioned data is the need to apply transformation and normalization procedures equally to all datasets.
```
Comment (SiminaB) [inline content/09.overfitting.md:23]:
```suggestion
One important caveat when working with partitioned data is the need to apply transformation and normalization procedures equally to all datasets.
```

File: content/09.overfitting.md (hunk 5 above)
Comment (SiminaB) [inline content/09.overfitting.md:21]:
```suggestion
Another way to reduce overfitting, as described by Chuang and Keiser, is to identify the baseline level of memorization that is occuring by training on data that has its labels randomly shuffled.
```

File: content/09.overfitting.md [hunk 7]
```diff
@@ -1,33 +1,36 @@
...
-Similarly, when dealing with sequence data, holding out data that are evolutionarily related or share structural homology to the training data can result in overfitting.
-In these cases, simply holding out test data selected from a random partition of the training data is insufficient.
-The best remedy for confounding variables is to [know your data](#know-your-problem) and to test your model on truly independent data.
+If overfitting is an issue, there are a variety of techniques to reduce overfitting, including data augmentation and regularization techniques such as dropout [@url:http://jmlr.csail.mit.edu/papers/v15/srivastava14a.html] and weight decay [@tag:krogh-weight-decay].
+Another way to reduce overfitting, as described by Chuang and Keiser, is to identify the baseline level of memorization that is occuring by training on data that has it labels randomly shuffled.
+By comparing the model performance with the shuffled data to that achieved with the actual data [@doi:10.1021/acschembio.8b00881], a practitioner can identify overfitting as a model that performs no better on real data, as this suggest that any predictive capacity is not due to data driven signal.
```
Comment (SiminaB) [inline content/09.overfitting.md:22]:
```suggestion
By comparing the model performance with the shuffled data to that achieved with the actual data [@doi:10.1021/acschembio.8b00881], a practitioner can identify overfitting as a model that performs no better on real data, as this suggest that any predictive capacity is not due to data-driven signal.
```

File: content/09.overfitting.md [hunk 8]
```diff
@@ -1,33 +1,36 @@
...
+One important caveat when working with paritioned data is the need to apply transformation and normalization procedures equally to all datasets.
+The parameters required for such procedures (for example, quantile normalization, a common standardization method when analyzing gene-expression data) should only be derived from the training data, and not from the tuning or test data.
+Additionally, many conventional metrics for classification (e.g. area under the receiver operating characteristic curve or AUROC) have limited utility in cases of extreme class imbalance [@pmid:25738806].
+Therefore, model performance should be evaluated with a carefully picked panel of relevant metrics that make minimal assumptions about the composition of the testing data [@doi:10.1021/acs.molpharmaceut.7b00578].
+
+When working with biological and medical data, one must also carefully consider potential sources of bias and/or non-independence when defining training and test sets.
+For example, a deep learning model for pneumonia detection in chest X-rays appeared to performed well, but then failed to generalize to outside hospitals [@doi:10.1371/journal.pmed.1002683].
```
Comment (SiminaB) [inline content/09.overfitting.md:29]:
```suggestion
For example, a deep learning model for pneumonia detection in chest X-rays appeared to performed well within the hospitals providing the training data, but then failed to generalize to other hospitals [@doi:10.1371/journal.pmed.1002683].
```

**Review by SiminaB — CHANGES_REQUESTED** [review]

File: content/09.overfitting.md [hunk 9]
```diff
@@ -1,33 +1,36 @@
...
+Just as a smarter student may have greater potential for memorization, deep learning models seem more prone to overfitting than traditional machine learning models.
 
-![A visual example of overfitting and failure to generalize. While a high-degree polynomial gets high accuracy on its training data, it performs poorly on data unlike that which it has seen before. In contrast, a simple linear regression works well on both datasets. The greater representational capacity of the polynomial is analogous to using a larger or deeper neural network.](images/overfitting.png){#fig:overfitting-fig}
+![A visual example of overfitting and failure to generalize. While a high-degree polynomial achieves high accuracy on its training data, it performs poorly on data with specificities that have not been seen before. That is, the model has learned the training dataset specifically rather than learning a generalizable pattern that represents data of this type. In contrast, a simple linear regression works well on both datasets. The greater representational capacity of the polynomial is analogous to using a larger or deeper neural network.](images/overfitting.png){#fig:overfitting-fig}
```
[the remark this answers is not in the document]
Comment (Benjamin-Lee) [reply content/09.overfitting.md:10]: Raised in #280

File: content/09.overfitting.md (hunk 4 above)
Comment (Benjamin-Lee) [reply content/09.overfitting.md:14]:
```suggestion
These three partitions allow us to optimize models by iterating between model learning (on the training set) and hyperparameter evaluation (on the tuning set) without affecting the final model assessment on the test set.
```

File: content/09.overfitting.md (hunk 5 above)
Comment (Benjamin-Lee) [reply content/09.overfitting.md]: I think its related, yes.

File: content/09.overfitting.md [hunk 10]
```diff
@@ -1,33 +1,36 @@
 ## Tip 7: Address deep neural networks' increased tendency to overfit the dataset {#overfitting}
 
-Overfitting is one of the most significant dangers you'll face in deep learning (and traditional machine learning).
-Put simply, overfitting occurs when a model fits patterns in the training data too closely, includes noise or non-scientifically relevant perturbations, or in the most extreme case, simply memorizes patterns in the training set.
+Overfitting is a challenge inherent to machine learning in general, and is one of the most significant challenges you'll face when applying deep learning specifically.
```
Comment (cgreene) [inline content/09.overfitting.md:3]:
```suggestion
Overfitting is a challenge inherent to machine learning in general, and is one of the most significant challenges one faces when applying deep learning.
```

File: content/09.overfitting.md [hunk 11]
```diff
@@ -1,33 +1,36 @@
...
+Overfitting occurs when a model fits patterns in the training data so closely that it is including non-generalizable noise or non-scientifically relevant perturbations in the relationships it is learning.
+In other words, it is as if the model is memorizing patterns in the specific data set it is training on rather than learning general relationships that hold across similar datasets.
```
Comment (cgreene) [inline content/09.overfitting.md:5]:
```suggestion
In other words, the model fits patterns that are overly specific to the data it is training on rather than learning general relationships that hold across similar datasets.
```

File: content/09.overfitting.md [hunk 12]
```diff
@@ -1,33 +1,36 @@
...
 This subtle distinction is made clearer by seeing what happens when a model is tested on data to which it was not exposed during training: just as a student who memorizes exam materials struggles to correctly answer questions for which they have not studied, a machine learning model that has overfit to its training data will perform poorly on unseen test data.
 Deep learning models are particularly susceptible to overfitting due to their relatively large number of parameters and associated representational capacity.
-To continue the student analogy, a smarter student has greater potential for memorization than average one and thus may be more inclined to memorize.
+Just as a smarter student may have greater potential for memorization, deep learning models seem more prone to overfitting than traditional machine learning models.
```
Comment (cgreene) [inline content/09.overfitting.md:8]:
```suggestion
Just as some students may have greater potential for memorization, deep learning models seem more prone to overfitting than machine learning models with fewer parameters.
```

File: content/09.overfitting.md [hunk 13]
```diff
@@ -1,33 +1,36 @@
...
-Model performance should be evaluated with a carefully picked panel of relevant metrics that make minimal assumptions about the composition of the testing data [@doi:10.1021/acs.molpharmaceut.7b00578], with particular consideration given to metrics that are most directly applicable to the task at hand.
-
-The simplest way to combat overfitting is to detect it.
-This can be done by splitting the dataset into three parts: a training set, a tuning set (also commonly called a validation set in the machine learning literature), and a test set.
-By exposing the model solely to the training data during fitting, a researcher can use the model's performance on the unseen test data to measure the amount of overfitting.
+In general, one of the most effective ways to combat overfitting is to detect it in the first place.
+One way to do this is to split the main dataset being worked on into three parts: a training set, a tuning set (also commonly called a validation set in the machine learning literature), and a test set.
```
Comment (cgreene) [inline content/09.overfitting.md:13]:
```suggestion
One way to do this is to split the main dataset being worked on into three independent parts: a training set, a tuning set (also commonly called a validation set in the machine learning literature), and a test set.
```

File: content/09.overfitting.md [hunk 14]
```diff
@@ -1,33 +1,36 @@
...
+These three partitions allow us to optimize models by iterating between model learning on the training set and hyperparameter evaluation on the tuning set without affecting the final model assessment on the test set.
+That is, the data used for testing should be "locked away" and used only once to evaluate the final model after all training and tuning steps are completed.
+A researcher can then use the model's performance on the unseen test data as a measure of how overfit (i.e. nongeneralizable) the model is.
```
Comment (cgreene) [inline content/09.overfitting.md:16]:
```suggestion
A researcher can then use the model's performance on the independent test data as a measure of how overfit (i.e. non-generalizable) the model is.
```

File: content/09.overfitting.md [hunk 15]
```diff
@@ -1,33 +1,36 @@
...
 
-Additionally, in biology and medicine it is critical to consider independence when defining training and test sets.
-For example, a deep learning model for pneumonia detection in chest X-rays performed well but failed to generalize to outside hospitals because they were able to detect which hospital the image was from and exploited this information when making predictions [@doi:10.1371/journal.pmed.1002683].
-Similarly, when dealing with sequence data, holding out data that are evolutionarily related or share structural homology to the training data can result in overfitting.
-In these cases, simply holding out test data selected from a random partition of the training data is insufficient.
-The best remedy for confounding variables is to [know your data](#know-your-problem) and to test your model on truly independent data.
+If overfitting is an issue, there are a variety of techniques to reduce overfitting, including data augmentation and regularization techniques such as dropout [@url:http://jmlr.csail.mit.edu/papers/v15/srivastava14a.html] and weight decay [@tag:krogh-weight-decay].
```
Comment (cgreene) [inline content/09.overfitting.md:20]:
```suggestion
If overfitting is an issue, there are a variety of techniques to reduce overfitting, including data augmentation and various regularization techniques [@url:http://jmlr.csail.mit.edu/papers/v15/srivastava14a.html; @tag:krogh-weight-decay].
```

File: content/09.overfitting.md [hunk 16]
```diff
@@ -1,33 +1,36 @@
...
+
+When working with biological and medical data, one must also carefully consider potential sources of bias and/or non-independence when defining training and test sets.
+For example, a deep learning model for pneumonia detection in chest X-rays appeared to performed well within the hospitals providing the training data, but then failed to generalize to other hospitals [@doi:10.1371/journal.pmed.1002683].
+This resulted from the deep learning model picking up on signal related to which hospital the images were from, and represents a type of artifact or "batch effect" that practitioners must be vigilant towards.
+When dealing with sequence data, holding out test data that are evolutionarily related or that share structural homology to the training data can result in overfitting that is hard to detect due to the inherent relatedness of the partitioned data (cite?).
+In such situations, simply holding out test data selected from a random partition of the training data can be insufficient.
+Again, the best remedy for identifying confounding variables is to [know your data](#know-your-problem) and to test your model on truly independent data.
```
Comment (cgreene) [inline content/09.overfitting.md:33]:
```suggestion
Again, the best remedy for identifying confounding variables is to [know your data](#know-your-problem) and to test models on truly independent data.
```

**Review by cgreene — APPROVED** [review]
A few thoughts and then go for it 🚀

**Review by Benjamin-Lee — APPROVED** [review]
